src/monodroid/jni/dylib-mono.h

```diff
@@ -137,6 +137,14 @@ inline MonoCounters& operator |= (MonoCounters& left, MonoCounters right)
 	typedef struct _MonoProfiler {} MonoProfiler;
 #endif
 
+#ifndef MonoProfilerDesc
+	typedef struct _MonoProfilerDesc {} MonoProfilerDesc;
+#endif
+
+#ifndef MonoProfilerHandle
+	typedef struct _MonoProfilerDesc* MonoProfilerHandle;
```

Comment:
This feels dubious, in that if `MonoProfilerDesc` *is* defined, then `struct _MonoProfilerDesc` might not be, because lines 140-142 won't exist.

Related: why would `MonoProfilerDesc` be defined?

For that matter, why wrap these in `#ifndef` in the first place?

Perhaps this line should instead be:

```suggestion
	typedef struct _MonoProfilerHandle {} *MonoProfilerHandle;
```

Reply:
Updated. I missed that the `MonoProfilerDesc` is not defined in mono. Only the `_MonoProfilerDesc` structure is used.

> For that matter, why wrap these in #ifndef in the first place?

IIRC, we used to build the host runtimes with mono headers. I guess it changed meanwhile?